Tasks/DeployAzureResourceGroup/ResourceGroup.ts

```diff
@@ -247,25 +254,32 @@ export class ResourceGroup {
     }
 
     public deleteResourceGroup() {
-        var armClient = new armResource.ResourceManagementClient(this.taskParameters.credentials, this.taskParameters.subscriptionId);
-        console.log(tl.loc("ARG_DeletingResourceGroup", this.taskParameters.resourceGroupName));
-        armClient.resourceGroups.deleteMethod(this.taskParameters.resourceGroupName,(error, result, request, response) => {
-            if (error) {
-                tl.setResult(tl.TaskResult.Failed, tl.loc("RGO_CouldNotDeletedResourceGroup", this.taskParameters.resourceGroupName, error.message));
-                process.exit();
-            }
-            tl.setResult(tl.TaskResult.Succeeded, tl.loc("RGO_DeletedResourceGroup", this.taskParameters.resourceGroupName));
-        });
+        var extDelPromise = this.MGExtensionManager.removeMGExtension();
+        extDelPromise.then((val) => {
```

Comment:
This promise is never rejected. In case the extension installation/removal fails, logging is done in the method postOperationCallBack